lib/dataset/bin_detail.h

```diff
@@ -6,16 +6,18 @@
 
 #include "scipp-dataset_export.h"
 #include "scipp/variable/subspan_view.h"
+#include "scipp/variable/util.h"
 #include "scipp/variable/variable.h"
 
 namespace scipp::dataset::bin_detail {
 
 template <class T> Variable as_subspan_view(T &&binned) {
   auto &&[indices, dim, buffer] = binned.template constituents<Variable>();
-  if constexpr (std::is_const_v<std::remove_reference_t<T>>)
+  if constexpr (std::is_const_v<std::remove_reference_t<T>>) {
     return subspan_view(std::as_const(buffer), dim, indices);
-  else
+  } else {
     return subspan_view(buffer, dim, indices);
+  }
 }
```

Comment:
Why was this file modified? Why the new include?

Reply:
I think there can be obscure ways to get a non-contiguous buffer, but I think we should ignore that, for now at least.